sys/phydat/phydat.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2018 Eistec AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_phydat
+ * @{
+ *
+ * @file
+ * @brief       Generic sensor/actuator data handling
+ *
+ * @author      Joakim Nohlgård <joakim.nohlgard@eistec.se>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include "phydat.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+uint8_t phydat_fit(phydat_t *dat, long value, unsigned int index, uint8_t prescale)
+{
+    assert(index < (sizeof(dat->val) / sizeof(dat->val[0])));
```

Comment:
This check is potentially safer towards future changes than just checking `PHYDAT_DIM`. However, I'm worried that people might be confused about this type of notation. Opinions?